### ISSUE TITLE
Bug: `sort-variable-declarations`: Missing dependency detections

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#256]
### Describe the bug

Issues related to https://github.com/azat-io/eslint-plugin-perfectionist/issues/254.

The following elements are not detected as dependencies and are considered invalid when they should not:

- [ ] Negation operator:
```
let b = 1,
    a = !b;
```

- [ ] Optional chained elements:
```
let b = {x: 1},
     a = b?.x;
```

- [ ] Non-null assertion operator:
```
let b = 1,
    a = b!;
```

- [ ] `as` expressions:
```
let b = 1,
    a = b as any;
```

- [ ] Type assertion expressions:
```
let b = 1,
    a = <any>b;
```

- [ ] All sorts of ternary expressions:
```
let b = 1,
    a = b ? 0 : 1;
```

- [ ] Template literal expressions:
```
let b = '',
    a = `${b}`
```

- [ ] Spread elements:
```
let b = {x: 1},
    a = {...b}
```

### Code example

See above

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

v3.4.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Fixed in 6beb536. Released in [v3.5.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.5.0).
